source/_docs/mu-plugin.md

```diff
@@ -0,0 +1,247 @@
+// Check if Drupal or WordPress is running via command line
+if (($_SERVER['REQUEST_URI'] == '/old') && (php_sapi_name() != "cli")) {
+  header('HTTP/1.0 301 Moved Permanently');
+  header('Location: https://'. $_SERVER['HTTP_HOST'] . '/new');
+
+  # Name transaction "redirect" in New Relic for improved reporting (optional)
+  if (extension_loaded('newrelic')) {
+    newrelic_name_transaction("redirect");
+  }
+
+  exit();
+}
+```
+
+### Cache Control
+Set `Cache-Control: max-age=0` by hooking into `send_headers`. This will override `max-age` configured within the Pantheon Cache plugin for all matching requests:
+
+```
+/*
+ * Set $regex_path_patterns accordingly.
+ *
+ * We don't set this variable for you, so you must define it
+ * yourself per your specific use case before the following conditional.
+ *
+ * For example, to exclude pages in the /news/ and /about/ path from cache, set:
+ *
+ *   $regex_path_patterns = array(
+ *     '#^/news/?#',
+ *     '#^/about/?#',
+ *   );
+ */
+
+
+// Loop through the patterns.
+foreach ($regex_path_patterns as $regex_path_pattern) {
+  if (preg_match($regex_path_pattern, $_SERVER['REQUEST_URI'])) {
+    add_action( 'send_headers', 'add_header_nocache', 15 );
+
+    // No need to continue the loop once there's a match.
+    break;
+  }
+}
+function add_header_nocache() {
+      header( 'Cache-Control: no-cache, must-revalidate, max-age=0' );
+}
+```
+
+### WP REST API (`wp-json`) Endpoints Cache
+For WP REST API endpoints, we can use the `rest_post_dispatch` filter and create a specific function to apply specific headers for each path or endpoint.
+
+```php
+/* For WP REST API specific paths, we use a different approach by using the rest_post_dispatch filter */
+
+// wp-json paths or any custom endpoints
+
+$regex_json_path_patterns = array(
+    '#^/wp-json/wp/v2/users?#',
+    '#^/wp-json/?#'
+);
+  
+foreach ($regex_json_path_patterns as $regex_json_path_pattern) {
+  if (preg_match($regex_json_path_pattern, $_SERVER['REQUEST_URI'])) {
+    // re-use the rest_post_dispatch filter in the Pantheon page cache plugin  
+    add_filter( 'rest_post_dispatch', 'filter_rest_post_dispatch_send_cache_control', 12, 2 );
+    // Re-define the send_header value with any custom Cache-Control header
+    function filter_rest_post_dispatch_send_cache_control( $response, $server ) {
+      $server->send_header( 'Cache-Control', 'no-cache, must-revalidate, max-age=0' );
+      return $response;
+    }
+    break;
+  }
+}
+```
+
+### Exclude Plugins for Redis cache
+A page load with 2,000 Redis calls can be 2 full seconds of object cache transactions. If a plugin you're using is erroneously creating a huge number of cache keys, you might be able to mitigate the problem by disabling cache persistence for the plugin's group. For example, to prevent [Autoptimize](https://wordpress.org/plugins/autoptimize/){.external} from using Redis:
+
+```php
+wp_cache_add_non_persistent_groups( array( 'autoptimize' ) );
```

Comment:
@JulienMelissas @carl-alberto @eabquina how you like me know?

Reply:
@alexfornuto Sorry I can't find a good example that implements this in Autoptimize and other caching plugins rarely define the plugin `group` variable so we might remove this part `For example, to prevent [Autoptimize](https://wordpress.org/plugins/autoptimize/){.external} from using Redis:
 ```php
wp_cache_add_non_persistent_groups( array( 'autoptimize' ) );`

We can then rather clarify this by:

```
For example, you have a plugin that sets the cache by:

wp_cache_set( 'cache_key', 'cache_data', 'my_plugin_group' );
wp_cache_get( 'cache_key', 'my_plugin_group' );

You can exclude that in Redis by:

wp_cache_add_non_persistent_groups( array( 'my_plugin_group' ) );
``

Reply:
As long as there is a solid example, I think that's fine! It's specific stuff, so I understand if finding an example is difficult.